pkg/virt-controller/watch/vm.go

```diff
@@ -162,7 +162,7 @@ func (c *VMController) execute(key string) error {
 	if !controller.ObservedLatestApiVersionAnnotation(vm) {
 		vm := vm.DeepCopy()
 		controller.SetLatestApiVersionAnnotation(vm)
-		_, err = c.clientset.VirtualMachine(vm.ObjectMeta.Namespace).Update(vm)
+		_, err = c.clientset.VirtualMachine(vm.ObjectMeta.Namespace).UpdateStatus(vm)
```

Comment:
what did you decide about updates/rollbacks that involve toggling on/off the status subresource?   Are you still thinking you'll attempt to set the Status with UpdateStatus, and if that fails attempt to use the old Update way? or something like that?

Reply:
Pushed the latest changes. It works like this:

 * virt-operator detects that the installed CRD has the subresource disabled, but the updated version has it enabled
 * It rolls out the new CRD without the status subresource enabled
 * then the control-plane is rolled over
 * it detects that the CRD needs the status subresource enabled to be in sync and does so